azurerm/resource_arm_dev_test_lab_schedule.go

```diff
@@ -0,0 +1,489 @@
+	}, nil
+
+}
+
+func flattenAzureRmDevTestLabScheduleRecurrenceWeekly(weeklyRecurrence *dtl.WeekDetails) []interface{} {
+	if weeklyRecurrence == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+
+	if weeklyRecurrence.Time != nil {
+		result["time"] = *weeklyRecurrence.Time
+	}
+
+	weekDays := make([]string, 0)
+	if w := weeklyRecurrence.Weekdays; w != nil {
+		weekDays = *w
+	}
+	result["week_days"] = weekDays
+
+	return []interface{}{result}
+}
+
+func expandArmDevTestLabScheduleRecurrenceHourly(d *schema.ResourceData) (*dtl.HourDetails, error) {
+	hourlyRecurrenceConfigs := d.Get("hourly_recurrence").(*schema.Set).List()
+	hourlyRecurrenceConfig := hourlyRecurrenceConfigs[0].(map[string]interface{})
+	hourlyMinute := int32(hourlyRecurrenceConfig["minute"].(int))
+
+	return &dtl.HourDetails{
+		Minute: &hourlyMinute,
+	}, nil
+
+}
+
+func flattenAzureRmDevTestLabScheduleRecurrenceHourly(hourlyRecurrence *dtl.HourDetails) []interface{} {
+	if hourlyRecurrence == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+
+	if hourlyRecurrence.Minute != nil {
+		result["minute"] = *hourlyRecurrence.Minute
+	}
+
+	return []interface{}{result}
+}
+
+func expandArmDevTestLabScheduleNotificationSettings(d *schema.ResourceData) (*dtl.NotificationSettings, error) {
+
+	notificationSettingsConfigs := d.Get("notification_settings").(*schema.Set).List()
+	notificationSettingsConfig := notificationSettingsConfigs[0].(map[string]interface{})
+	webhookUrl := notificationSettingsConfig["webhook_url"].(string)
+	timeInMinutes := int32(notificationSettingsConfig["time_in_minutes"].(int))
+
+	var notificationStatus dtl.NotificationStatus
+	switch status := notificationSettingsConfig["status"]; status {
+	case string(dtl.NotificationStatusEnabled):
+		notificationStatus = dtl.NotificationStatusEnabled
+	case string(dtl.NotificationStatusDisabled):
+		notificationStatus = dtl.NotificationStatusDisabled
+	default:
+	}
+
+	return &dtl.NotificationSettings{
+		WebhookURL:    &webhookUrl,
+		TimeInMinutes: &timeInMinutes,
+		Status:        notificationStatus,
+	}, nil
+
+}
+
+func flattenAzureRmDevTestLabScheduleNotificationSettings(notificationSettings *dtl.NotificationSettings) []interface{} {
+	if notificationSettings == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+
+	if notificationSettings.WebhookURL != nil {
+		result["webhook_url"] = *notificationSettings.WebhookURL
+	}
+
+	if notificationSettings.TimeInMinutes != nil {
+		result["time_in_minutes"] = *notificationSettings.TimeInMinutes
+	}
+
+	if string(notificationSettings.Status) != "" {
+
+		var notificationStatus string
+		switch notificationSettings.Status {
+		case dtl.NotificationStatusEnabled:
+			notificationStatus = string(dtl.NotificationStatusEnabled)
+		case dtl.NotificationStatusDisabled:
+			notificationStatus = string(dtl.NotificationStatusDisabled)
+		default:
+		}
+		result["status"] = notificationStatus
```

Comment:
we can replace this switch statement with:

```suggestion
		result["status"] = string(notificationSettings.Status)
```

Reply:
Done.